syft/pkg/cataloger/javascript/parse_node_binary.go

```diff
@@ -38,6 +38,5 @@ func parseNodeBinary(_ source.FileResolver, _ *generic.Environment, reader sourc
 		p.Language = pkg.JavaScript
 		return []pkg.Package{*p}, nil, nil
 	}
-	p.SetID()
```

Comment:
Good catch! We still want this `p.SetID()`, but as you pointed out, it should be moved after line 38, so:

```go
p.Language = pkg.JavaScript
p.SetID()
return []pkg.Package{*p}, nil, nil
```

Reply:
I've adapted the merge request based on hint.